llvm/lib/CodeGen/StackMaps.cpp

```diff
@@ -305,8 +305,22 @@ StackMaps::parseOperand(MachineInstr::const_mop_iterator MOI,
     // each one of them. Note, that the stackmap may track either of %rbx or
     // %rcx, resulting in different ways below to retrieve the mappings.
     int ExtraReg = 0;
+    Register R = MOI->getReg();
     if (MOI->isReg()) {
-      Register R = MOI->getReg();
+      if (MOI->isKill()) {
+        // There's no point in tracking a killed register. Instead, try and
+        // replace it with another register with the same value (possibly due
+        // to spilling), to increase the likelyhood off us tracking that value.
```

Comment:
We discussed this the other day, but I wouldn't say a temporary save/restore from one register to another is a "spill".

You said LLVM uses this term that way though? Is that so? Can you show us?

Reply:
Can't find it anymore and don't really care enough to keep looking. Happy for you to suggest a different wording if you dislike the term spilling to describe storing one register in another to free the first one up.

Reply:
you can just say the value was moved to another register. There are many reasons why that could happen, the "spill" being just one of them.

Reply:
[bedd427](https://github.com/ykjit/ykllvm/pull/105/commits/bedd4273722a74a617f912e3630e3876ab8d4024)